### ISSUE TITLE
Autoupdater asking for credentials

### ISSUE THREAD

**Issue by DaveMtb** [sparkle-project/Sparkle#1470]
### Description of the problem

We're seeing an issue where Sparkle is asking for credentials (the Autoupdater asks for your password). We haven't seen this before, but it now seems to occur if we notarize the app prior to pushing it to the update site.

Our big problem with it asking for permission is that our archive file consists of a PKG file. If Sparkle doesn't ask for permissions, then the PKG file is run and everything proceeds normally. 
If Sparkle DOES ask for permissions, then the PKG file is NOT run. However the app still installs. The problem is that post-install scripts in the PKG file don't appear to be run, which causes problems since those scripts are important.

Looking at the console log, if I go to the cached (PersistentDownloads area) and run the PKG file, it runs just fine.

We just started seeing this recently, and we're not sure as to what kind of change might have caused it. Any help would be appreciated.

### Do you use Sandboxing in your app?
No.

### Version of `Sparkle.framework` in the latest version of your app
1.21.3

### Version of `Sparkle.framework` in the old version of app that your users have (or N/A)
1.21.3

### Sparkle's output from Console.app
```
default	14:08:28.141085 -0400	XXXXX	Extracting using '/usr/bin/ditto' '-x' '-k' '-' < '/Users/dYYYYY/Library/Caches/com.XXXXX.XXXXX/org.sparkle-project.Sparkle/PersistentDownloads/Uwd2f35HN/XXXXX 19.2.0.0.20190916.1/XXXXX.v19.2.0.0.Commercial.zip' '/Users/dYYYYY/Library/Caches/com.XXXXX.XXXXX/org.sparkle-project.Sparkle/PersistentDownloads/Uwd2f35HN/XXXXX 19.2.0.0.20190916.1'
default	14:08:31.353653 -0400	XXXXX	OK: EdDSA signature is correct
default	14:08:31.371536 -0400	lsd	Non-fatal error enumerating at <private>, continuing: Error Domain=NSCocoaErrorDomain Code=260 "The file “PlugIns” couldn’t be opened because there is no such file." UserInfo={NSURL=PlugIns/ -- file:///Users/dYYYYY/Library/Caches/com.XXXXX.XXXXX/org.sparkle-project.Sparkle/XXXXX%20(Autoupdate).app/, NSFilePath=/Users/dYYYYY/Library/Caches/com.XXXXX.XXXXX/org.sparkle-project.Sparkle/XXXXX (Autoupdate).app/PlugIns, NSUnderlyingError=0x7fe265d8ac60 {Error Domain=NSPOSIXErrorDomain Code=2 "No such file or directory"}}
default	14:08:31.380081 -0400	lsd	Non-fatal error enumerating at <private>, continuing: Error Domain=NSCocoaErrorDomain Code=260 "The file “PlugIns” couldn’t be opened because there is no such file." UserInfo={NSURL=PlugIns/ -- file:///Users/dYYYYY/Library/Caches/com.XXXXX.XXXXX/org.sparkle-project.Sparkle/XXXXX%20(Autoupdate).app/Contents/, NSFilePath=/Users/dYYYYY/Library/Caches/com.XXXXX.XXXXX/org.sparkle-project.Sparkle/XXXXX (Autoupdate).app/Contents/PlugIns, NSUnderlyingError=0x7fe267014a20 {Error Domain=NSPOSIXErrorDomain Code=2 "No such file or directory"}}
default	14:08:31.430054 -0400	lsd	Non-fatal error enumerating at <private>, continuing: Error Domain=NSCocoaErrorDomain Code=260 "The file “PlugIns” couldn’t be opened because there is no such file." UserInfo={NSURL=PlugIns/ -- file:///Users/dYYYYY/Library/Caches/com.XXXXX.XXXXX/org.sparkle-project.Sparkle/XXXXX%20(Autoupdate).app/Contents/, NSFilePath=/Users/dYYYYY/Library/Caches/com.XXXXX.XXXXX/org.sparkle-project.Sparkle/XXXXX (Autoupdate).app/Contents/PlugIns, NSUnderlyingError=0x7fe265f1e310 {Error Domain=NSPOSIXErrorDomain Code=2 "No such file or directory"}}
default	14:08:31.543446 -0400	launchservicesd	CHECKIN:0x0-0x7affaf8 1812 org.sparkle-project.Sparkle.Autoupdate
default	14:08:31.621595 -0400	Autoupdate	Current system appearance, (HLTB: 1), (SLS: 0)
default	14:08:31.627481 -0400	Autoupdate	Post-registration system appearance: (HLTB: 1)
default	14:08:31.640843 -0400	distnoted	register name: com.apple.xctest.FakeForceTouchDevice object: org.sparkle-project.Sparkle.Autoupdate token: f4263 pid: 1812
default	14:08:31.657436 -0400	Autoupdate	PID to listen: 1804
default	14:08:34.582358 -0400	Autoupdate	27366555: RECEIVED OUT-OF-SEQUENCE NOTIFICATION: 0 vs 2457, 513, <private>
default	14:08:35.168161 -0400	Autoupdate	NSApp cache appearance:
-NSRequiresAquaSystemAppearance: 0
-appearance: (null)
-effectiveAppearance: <NSCompositeAppearance: 0x7f9667a04250
 (
    "<NSAquaAppearance: 0x7f966790dbd0>",
    "<NSSystemAppearance: 0x7f9667a04d50>"
)>
default	14:08:35.192929 -0400	distnoted	register name: com.apple.nsquiet_safe_quit_give_reason object: org.sparkle-project.Sparkle.Autoupdate token: f4294 pid: 1812
default	14:08:35.285307 -0400	Autoupdate	Sparkle: Authorization required to run pkg installer file:///Users/dYYYYY/Library/Caches/com.XXXXX.XXXXX/org.sparkle-project.Sparkle/PersistentDownloads/Uwd2f35HN/XXXXX%2019.2.0.0.20190916.1/XXXXX.v19.2.0.0.Commercial.pkg
default	14:08:40.594063 -0400	distnoted	register name: com.apple.sharedfilelist.change object: com.apple.LSSharedFileList.ApplicationRecentDocuments/org.sparkle-project.sparkle.autoupdate token: f5356 pid: 728
default	14:08:41.659874 -0400	Autoupdate	LSExceptions shared instance invalidated for timeout.
```

### Steps to reproduce the behavior

We can sometimes get it to work by not notarizing the app, but I'm unclear if that's just luck. We've been trying multiple approaches, but the results are sometimes inconsistent.

**Comment by kornelski** [comment]
The log says "Sparkle: Authorization required to run pkg installer"

This is exepcted behavior. If your update is in `.pkg` format, it will always require authorization.

To have updates without authorization you must stop using `.pkg` and switch to `.app` in a zipfile or similar.

**Comment by DaveMtb** [comment]
Thanks for the reply. However, that's not been our experience. We've been using pkg formats for years. They _have_ required authorization, but after the the pkg file was running, never before. 

Even so, having the Autoupdater prompt for a password would be acceptable, except in that case it never seems to run the pkg file. No UI from the pkg file is ever displayed. The app _is_ installed, but the post-install scripts don't seem to get run. 

The last time we tested this was in July and it worked as we expected (no Autoupdater prompt, and the pkg file was run). Something changed since then, but I'm not sure what. We can _sometimes_ get it to work, but the results are inconsistent.

Any thoughts?

**Comment by kornelski** [comment]
You can turn the installer UI back on if you want:

https://sparkle-project.org/documentation/package-updates/

**Comment by DaveMtb** [comment]
That was the missing piece. We have dedicated build machines with lots of scripts involved. When the build team made changes related to notarization, they changed the name of the pkg file that gets zipped for the archive. All of sudden results from the build machine stopped working, while manual builds we did still worked for the most part.

Seeing your link refreshed my memory, and we hunted down the problem with the build scripts.

Thanks for the help!
